core-web/libs/block-editor/src/lib/extensions/ai-image-prompt/ai-image-prompt.store.ts

```diff
@@ -0,0 +1,131 @@
+
+    readonly isLoading$ = this.select(
+        this.state$,
+        ({ status }) => status === ComponentStatus.LOADING
+    );
+
+    readonly getContentlets$ = this.select(this.state$, ({ contentlets }) => contentlets);
+
+    readonly setPromptType = this.updater((state, selectedPromptType: PromptType) => ({
+        ...state,
+        selectedPromptType
+    }));
+
+    readonly showDialog = this.updater((state, editorContent: string) => ({
+        ...state,
+        showDialog: true,
+        selectedPromptType: DEFAULT_INPUT_PROMPT,
+        editorContent
+    }));
+
+    readonly hideDialog = this.updater((state) => ({
+        ...state,
+        showDialog: false,
+        selectedPromptType: null
+    }));
+
+    readonly vm$: Observable<VmAiImagePrompt> = this.select(
+        this.state$,
+        ({ selectedPromptType, showDialog, status }) => ({
+            selectedPromptType,
+            showDialog,
+            status
+        })
+    );
+
+    /**
+     * The `generateImage` variable is a function that generates an image based on a given prompt.
+     *
+     * @param {Observable<string>} prompt$ - An observable representing the prompt.
+     * @returns {Observable} - An observable that emits the generated image.
+     */
+    readonly generateImage = this.effect((prompt$: Observable<string>) => {
+        return prompt$.pipe(
+            withLatestFrom(this.state$),
+            switchMap(([prompt, { selectedPromptType, editorContent }]) => {
+                const cleanPrompt = prompt ? prompt.trim() : '';
+
+                const finalPrompt =
+                    selectedPromptType === 'auto' && editorContent
+                        ? `${cleanPrompt} to illustrate the following content: ${editorContent}`
+                        : cleanPrompt;
+                this.patchState({ status: ComponentStatus.LOADING, prompt: finalPrompt });
+
+                return this.dotAiService.generateAndPublishImage(finalPrompt).pipe(
+                    tapResponse(
+                        (contentLets) => {
+                            this.patchState({
+                                status: ComponentStatus.IDLE,
+                                contentlets: contentLets
+                            });
+                        },
+                        () => {
+                            this.patchState({ status: ComponentStatus.IDLE });
+                        }
+                    )
+                );
+            })
+        );
+    });
+
+    /**
+     * Regenerate the image and call the generateImage effect with withLatestFrom prompt
+     */
+    readonly reGenerateContent = this.effect((trigger$: Observable<void>) => {
+        return trigger$.pipe(
+            withLatestFrom(this.state$),
+            map(([_, { prompt }]) => {
+                if (!prompt) return;
+                this.generateImage(of(prompt));
+            })
```

Comment:
Shouldn't this be a `tap`? If there is a `prompt` we are not returning a value

Reply:
fixed